Da-Bois-Tracking/Tracking-App
# Ta 8 incorporate swagger open api

### In this PR, I'm adding/changing the following:

- Implementing Swagger UI for API Documentation

### Please do the following to test my code:

1. Make sure your local main is up to date
   `git checkout main`
   `git pull`
2. Checkout to my feature branch
   `git checkout TA-8-Incorporate-Swagger-OpenAPI`
3. Activate your virtual environment
   - For Windows:
      `python -m venv venv`
      `venv/Scripts/activate`
   - For macOS / Linux:
      `python3 -m venv venv`
      `source venv/bin/activate`
4. Install Dependencies:
   `pip install -r requirements.txt`

5. Start the Flask Application:
   - Navigate to the project directory `/backend` and run:
      `python app.py`
6. In your browser, navigate to `localhost:5000/api/docs`

7. Verify the Swagger UI
   - You should see the following:
      - [ ] Title at the top of the page that says "Tracker App API"
      - [ ] A description stating "API for Tracker App"
      - [ ] A GET request that leads to the endpoint `/hello`
      
![image](https://github.com/Da-Bois-Tracking/Tracking-App/assets/57964857/2e7b81c8-022e-4d37-bbba-f32dc8f65cd4)


8. TEST
   - Click on the GET request for `/hello` to open up the dropdown. 
![image](https://github.com/Da-Bois-Tracking/Tracking-App/assets/57964857/5124fed4-0f89-4210-b244-c286c51fd2ab)
   - Press on `Try it out`, then `Execute`
   - Upon execution, you should see a Code 200 return along with a response body of:
      `{ "message": "Hello from Flask!" }
![image](https://github.com/Da-Bois-Tracking/Tracking-App/assets/57964857/12cd176d-4e27-4d71-8291-47a0f8680661)


[Related Jira Ticket](https://bois-tracking-app.atlassian.net/browse/TA-8)
